packages/notebook/src/actions.tsx

```diff
@@ -652,7 +659,10 @@ export namespace NotebookActions {
       notebook.activeCellIndex++;
     }
 
-    void Private.handleRunState(notebook, state, true);
+    // If a cell is outside of viewport and scrolling is needed,
+    // prefer to align the cell to the top viewport edge,
+    // rather than to the bottom (so that editor is visible)
+    void Private.handleRunState(notebook, state, true, 'center');
```

Comment:
Why 'center' and not 'top-center'? I thought the idea was to keep the cell that we just left in view, which we cannot guarantee if we scroll the center of the next cell into the center of the viewport.

Reply:
Great question! This is because it only applies to `auto` strategy and `handleRunState` uses the `smart` strategy, which will:
- (a) not scroll if significantly visible, otherwise
- (b) scroll using `auto` heuristics if less than one viewport away
        this point is important because if an item is less than one viewport and not significantly visible it implies that it fits in the viewport hence we can apply `center` rather than `top-center`.
- (c) choose between `top-center` and `center` depending on item size.

So in (b) we can use `center` over `top-center`; but should we? For small cells it is nicer, and in a rare case of item size being equal to viewport we loose past context (future context would not be visible either way), so maybe we could choose between `center` and `top-center` depending on item size, but I did not want to make the logic any more complicated; in fact I was considering removing this fallback to `auto` heuristic here but was also cautious not to modify the logic too much in case if downstream already depends on `smart` falling back to `auto` in this circumstance. In either case, yes we _could_ also use `top-center` here. What do you think?

This also highlights that the comment above the line you commented on is outdated because I had not updated it since the previous solution was in place. I will update it now.

Reply:
The comment was updated in 5b02962df509818e7cc5ca0b8c6515a4c74f03fa

Reply:
I can't help but wondering if the added code complexity outweighs the UX benefit. Seems to me that if we don't introduce the `alignPreference` parameter then the consequence is that in the scenario labeled (b), in which the cell fits within the viewport and is less than one viewport away, then the cell will be scrolled to in such a way that either its top edge will align with the top edge of the scrollable container or its bottom edge with the bottom edge. I'm probably missing something but I don't see how that's significantly worse than scrolling the cell's top edge (or center) to the center of the scroll-container.